### DISCUSSION TITLE
#45 Optimizer #49 6:59:00 Loss-Test loss

### DISCUSSION THREAD

**Discussion by niktheod** [mrdbourke/pytorch-deep-learning#320] [General]
Hi
I have 2 questions regarding the 01 chapter.
First of all I dont get how the optimizer knows how to change the value of the parameters since it doesnt have any connection with the loss function. I mean how does it know what the value of the loss function is since it is not a parameter of the optimizer?
                            
And my second question is about the Loss and the Test loss that were calculated at around 6:59:00.
I dont get why they are so different. Shouldnt they be the same since our model makes predictions by using the same parameters for both of them(self.weight and self.bias)? 
Thanks in advance!

**Comment by Janko960** [comment]
I have just started the course so maybe I'm wrong but I understand it as follows:
1. You pass input through the network in a forward pass and you get an output.
2. Loss function calculates the error based on predicted output vs true output.
3. `loss.backward()` - propagates the error through the network using a chain rule (partial differential equations). I assume that the network stores backpropagated gradients(errors) at each layer for learnable parameters
4. When initiating the optimizer, parameters of the networks are set, so the optimizer, based on backpropagated gradients calculates new weights(parameters).

If its wrong please correct.

P.S. English is not my native language.
